### PR TITLE
Don't search again if someone searched nearby recently.

### PR DESCRIPTION
**Pull request by jhugman**
This pull request introducing the recording of a search as a geo-coord, a geohash and a timestamp.

It also introduces code to construct firebase queries to check if a given lat lon falls into a given geo-circle.

Finally, when a new search request comes in, a geocircle around that search is checked for other searches performed. If any search record is found, the timestamp is checked. If the time of the previous search is longer than an expiry time then the previous search record is deleted, and the new search can continue. Otherwise, a fresh search can be done.

### CODE REVIEW

File: app/request_handler.py [hunk 1]
```diff
@@ -37,10 +37,42 @@ def writeVenueRecord(biz, details, idObj = None):
...
+
+    for query in queries:
+        results = db.child(searchesTable).order_by_key().start_at(query[0]).end_at(query[1]).get()
+        for result in results.each():
+            record = result.val()
+            if record.get("time", 0) + searchCacheExpiry < now:
+                db.child(searchesTable).child(result.key()).remove()
+                continue
+            # double check that we're within distance
+            circleDistance = geo.distance(center, record["l"]) * 1000
```
Comment (liuche) [inline app/request_handler.py:64]: Is this 1000 hard-coded/and should use radius instead?

**Review by liuche — APPROVED** [review]
Just the one comment - also, maybe adding a comment for the ported code linking back to the original sources.

Other than that, feel free to merge!

File: app/request_handler.py (hunk 1 above)
Comment (jhugman) [reply app/request_handler.py:64]: Ah, no. `geo.distance()` returns in km. 

We need this in metres.

**Comment by jhugman** [comment]
Added clarifying comments.

**Pull request merged**